### PR TITLE
GasPriceMinimum Foundry Migration

### LINKED ISSUE #10803: test/common/gaspriceminimum.ts

### PR DESCRIPTION
**Pull request by soloseng**
### Description

Migration of the GasPriceMinimum contracts tests to foundry.

### Other changes

- Added `v4.4.2` version of openzeppelin submodule.
 
- Added latest version of `celo-foundry` that supports solidity contracts with `pragma 0.8.0`

### Tested

uint tested

### Related issues

- Fixes #10803

### CODE REVIEW

File: packages/protocol/test-sol/common/GasPriceMinimum.t.sol [hunk 1]
```diff
@@ -0,0 +1,272 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.7 <0.8.20;
+
+import "celo-foundry-8/Test.sol";
+
+import "../../contracts/common/FixidityLib.sol";
```
Comment (martinvol) [inline packages/protocol/test-sol/common/GasPriceMinimum.t.sol]: include the remapping

File: packages/protocol/test-sol/common/GasPriceMinimum.t.sol [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+      gasPriceMinimumFloor,
+      targetDensity,
+      adjustmentSpeed,
+      0
+    );
+  }
+}
+
+contract GasPriceMinimumSetAdjustmentSpeed is GasPriceMinimumTest {
+  using FixidityLib for FixidityLib.Fraction;
```
Comment (martinvol) [inline packages/protocol/test-sol/common/GasPriceMinimum.t.sol:90]: add this to parent?

File: packages/protocol/test-sol/common/GasPriceMinimum.t.sol [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    uint256 currentGasPriceMinimum = gasPriceMinimum.gasPriceMinimum();
+
+    gasPriceMinimum.setGasPriceMinimumFloor(1);
+
+    uint256 expectedUpdatedGasPriceMinimum = (currentGasPriceMinimum * 3) / 4 + 1;
+
+    assertEq(gasPriceMinimum.getUpdatedGasPriceMinimum(0, 1), expectedUpdatedGasPriceMinimum);
+  }
+
+  function test_shouldReturnAnUpdatedGasPriceMinimumThatMatchesATypescriptImplementation_WhenTheFullnessOfTheBlockIsRandom()
```
Comment (martinvol) [inline packages/protocol/test-sol/common/GasPriceMinimum.t.sol]:
```suggestion
  function test_shouldReturnAnUpdatedGasPriceMinimumThatMatchesARandomNumber_WhenTheFullnessOfTheBlockIsRandom()
```

**Review by martinvol — APPROVED** [review]

File: packages/protocol/test-sol/common/GasPriceMinimum.t.sol (hunk 2 above)
Comment (soloseng) [reply packages/protocol/test-sol/common/GasPriceMinimum.t.sol:90]: I believe that in `pragma ^0.8` it is required to reimport libraries `using A for B;` in each contract.

> The using A for B; directive is active only within the current scope (either the contract or the current module/source unit), including within all of its functions, and has no effect outside of the contract or module in which it is used.

https://docs.soliditylang.org/en/v0.8.15/contracts.html#using-for

**Pull request merged**